src/core/sys/windows/stacktrace.d

```diff
@@ -294,7 +294,19 @@ private:
             size_t decodeIndex = 0;
             tempSymName = decodeDmdString(tempSymName, decodeIndex);
         }
-        res ~= demangle(tempSymName, demangleBuf);
+        auto demangledName = demangle(tempSymName, demangleBuf);
+
+        version (Shared)
```

Comment:
Why limit this to a shared build of the runtime?

Reply:
you're right, for Windows it is not necessary

Reply:
Why is it necessary on other platforms. "Shared" is not a version for dlopen support, but whether the runtime itself is a shared library.

Reply:
druntime must be launched with -ldl, otherwise the linker will not find dlopen.
I don't know how i can check this in a better way

Reply:
The compiler already adds -ldl to the linker command line (at least for linux):
https://github.com/dlang/dmd/blob/e8f984995598543279c07b9800d19a9e19321562/src/dmd/link.d#L756

Reply:
Yes, but only if phobos is linked

Reply:
Ok but if I create the object files with dmd and then link it manually? I'm not sure it's a good idea to depend on libdl for druntime

Reply:
Ah, I see that `rt_loadLibrary` is only implemented if `version(Shared)`, so it's consistent to not use it here, too. That restriction seems a bit strange, though.

Reply:
We could do better if dmd supports weak symbols.
It would be enough to create a dlopen function that returns null marked as weak symbol, so if the user links libdl the real dlopen would be used